refact(occupation): refactoring of occupation.dm

O0oMaximo0O opened the pull request:
Удалил из файла упоминания ранга и ветви.
Почистил все образовавшиеся после этого хвосты.
Порефакторил функции, дал более выражающие суть их работы имена(на мой взгляд, готов выслушать ваши предложения).
Заменил "магические числа" на дефайны, добавил для них новый файл `job.dm` в `__defines`.

В дальнейшем планирую выполнить изначальную цель ПР-а - пофиксить отображение профы ассистента в лобби.
Для этого пока что самое простое решение - сделать его профессией высокого приоритета вместо низкого. Но это будет сбрасывать другую профу, выбранную с высоким приоритетом, чего хотелось бы(вероятно) избежать.
Также это не убирает несколько других мест, где ассистент обрабатывается особо. С удовольствием выслушаю предложения по исправлению этого.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

O0oMaximo0O committed:
fix(lobby): fixed show of "Assistant" occupation

O0oMaximo0O committed:
fix

O0oMaximo0O committed:
tweak(lobby): now subjob is being shown

HonkyDonky commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -73,8 +73,12 @@
 				var/highjob
-				if(player.client && player.client.prefs && player.client.prefs.job_high)
-					highjob = " as [player.client.prefs.job_high]"
+				if(player.client && player.client.prefs)
+					if(player.client.prefs.job_low && ("Assistant" in player.client.prefs.job_low))
+						highjob = " as Assistant"
+					else
+						if(player.client.prefs.job_high)
+							highjob = " as [player.client.prefs.job_high]"
```
Честно? Выглядит страшно. Уверен, что можно сделать короче и красивее

А и, приглядись к оператору `?.`, чтобы заместо `if(player.client && player.client.prefs)` было `if(player.client?.prefs)`. Это, по сути, простое сокращение. `A && A.B` == `A?.B`

TobyThorne replied:
Не просто сокращение, а ещё и оптимизация, к слову. Нанооптимизация, правда, но всё же.

O0oMaximo0O committed:
Update new_player.dm

TermService commented:
слишком костыльно, надо переделать файл _code/modules/client/preference_setup/occupation.dm_ в нормальный читаемый вид и исправить логику работы в нём

O0oMaximo0O commented:
Окей, попробую

O0oMaximo0O committed:
refact(occupation): refactoring of occupation.dm

Removed `military rank` and `military branch`.
Functions renamed to more suitable names for its results.
`Magic numbers` replaced by defines, they are placed into
`__defines/job.dm`.

O0oMaximo0O committed:
Merge branch 'fix-occupation-in-lobby' of github.com:O0oMaximo0O/OnyxBay into fix-occupation-in-lobby

kreeperHLC commented:
Конфликты

O0oMaximo0O commented:
Ну клёва, чё. Пойду разрешать

O0oMaximo0O commented:
Месяц один аппрув висел...

O0oMaximo0O committed:
Merge branch 'dev' into fix-occupation-in-lobby

O0oMaximo0O commented:
А, не, уже почти 2 месяца аппрув висел...
